apps/examples/wifi_test/wifi_test.c

```diff
@@ -204,10 +204,10 @@ int wifi_test_proc(int argc, char *argv[])
 		memset(payload, 0, BUF_MAX_SIZE);
 
 		printScanResult(scan_info);
-		snprintf(payload, BUF_MAX_SIZE, "/%s|", DEVICE_NAME);
+		snprintf(payload, BUF_MAX_SIZE - 1, "/%s|", DEVICE_NAME);
 		for (i = 0; i < aps_cnt; i++) {
-			sprintf(buf, "%02d:%d/", g_aps[i].id, g_aps[i].rssi);
-			strncat(payload, buf, strlen(buf));
+			snprintf(buf, BUF_MAX_SIZE - 1, "%02d:%d/", g_aps[i].id, g_aps[i].rssi);
+			strncat(payload, buf, (BUF_MAX_SIZE - strlen(payload)) > strlen(buf) ? strlen(buf) : BUF_MAX_SIZE - strlen(payload) - 1);
```

Comment:
how about to drop the payload if buffer is overflow

Reply:
I modify this ad you recommend.